flycheck.el

```diff
@@ -2913,7 +3024,8 @@ variables of Flycheck."
   (dolist (buffer (buffer-list))
     (with-current-buffer buffer
       (when flycheck-mode
-        (flycheck-teardown)))))
+        (flycheck-teardown))))
+  (remove-hook 'buffer-list-update-hook #'flycheck-handle-buffer-switch))
```

Comment:
So we add the hook anytime `flycheck-mode` is activated, but the only time we remove it is when Emacs exits?

Obviously, we should remove the hook when we kill the last flycheck buffer, right?  @cpitclaudel is there an idiomatic way of cleaning up in that situation, or do we have to iterate through the buffer-list or keep track of the number of flycheck-mode buffers?

Reply:
We have to iterate, AFAIK.  Just keeping a count would probably be OK, but it's more brittle than needed; just going over the list of buffers should be good.
There's a cheap function `buffer-local-value`, that we can use to check many buffers quickly.